### PR TITLE
FIO.proto ref_block_prefix type changed to uint64

### PR DESCRIPTION
**Pull request by hewigovens**
Quick fix for #878

### CODE REVIEW

**Comment by catenocrypt** [comment]
I think allowing a larger type here does not make much sense, see my comment in the Issue.  However, we could allow larger type for blockNum and to truncation inside walletcore, there the spec explicitly says that module 0x10000 has to be taken.
 
See comment in  #878 
https://github.com/trustwallet/wallet-core/issues/878#issuecomment-593263113
.

**Comment by hewigovens** [comment]
https://github.com/trustwallet/wallet-core/issues/878#issuecomment-593265849

**Comment by catenocrypt** [comment]
I did some adjustments:
- added tests for the range (encodes transaction, parses out parameters, and compares chain params with expected)
- moved truncation to Transaction class
- extended to BlockNumber as well.

File: src/FIO/Transaction.cpp [hunk 1]
```diff
@@ -5,14 +5,21 @@
...
+    expiration = expiryTime;
+    refBlockNumber = static_cast<uint16_t>(chainParams.headBlockNumber & 0xffff);
+    refBlockPrefix = static_cast<uint32_t>(chainParams.refBlockPrefix & 0xffffffff);
+}
+
 void Transaction::serialize(Data& out) const {
     encode32LE(expiration, out);
     encode16LE(refBlockNumber, out);
-    encode32LE(refBlockPrefix, out);
+    encode32LE(static_cast<uint32_t>(refBlockPrefix), out);
```
Comment (hewigovens) [inline src/FIO/Transaction.cpp]: it looks like we can revert this line? @catenocrypt

Comment (catenocrypt) [reply src/FIO/Transaction.cpp]: true, done.

**Pull request merged**